src/API/Site/Controllers/MerchantCenter/SyncableProductsCountController.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseOptionsController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\JobRepository;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\UpdateSyncableProductsCount;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SyncableProductsCountController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class SyncableProductsCountController extends BaseOptionsController {
+
+	/**
+	 * @var JobRepository
+	 */
+	protected $job_repository;
+
+	/**
+	 * SyncableProductsCountController constructor.
+	 *
+	 * @param RESTServer    $server
+	 * @param JobRepository $job_repository
+	 */
+	public function __construct( RESTServer $server, JobRepository $job_repository ) {
+		parent::__construct( $server );
+		$this->job_repository = $job_repository;
+	}
+
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		$this->register_route(
+			'mc/syncable-products-count',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_syncable_products_count_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->update_syncable_products_count_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the callback function for marking setup complete.
+	 *
+	 * @return callable
+	 */
+	protected function get_syncable_products_count_callback(): callable {
+		return function( Request $request ) {
+			$response = [
+				'count' => null,
+			];
+
+			$count = (int) $this->options->get( OptionsInterface::SYNCABLE_PRODUCTS_COUNT );
+
+			if ( $count ) {
+				$response['count'] = $count;
+			}
```

Comment:
Thanks, this is quite a bug. Updated in 963b327 and 13e1a57. The former updated the job `UpdateSyncableProductsCount` so it will store `0` in the database when there is no syncable products, also added tests for unsyncable products. The latter updated the controller to return `0` when there is no syncable products, and return `null` when the job hasn't finished.